clients/unshaded/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -297,19 +299,24 @@ public synchronized int createDependency(List<String> parents, List<String> chil
    * @param path The path of the file
    * @param ufsPath The path of the file in the under file system. If this is empty, the file does
    *        not exist in the under file system yet.
-   * @param blockSizeByte The size of the block in bytes. It is -1 iff ufsPath is non-empty.
+   * @param blockSizeBytes The size of the block in bytes. It is -1 iff ufsPath is non-empty.
    * @param recursive Creates necessary parent folders if true, not otherwise.
    * @return The file id, which is globally unique
    * @throws IOException if the underlying master RPC fails
    */
   @Override
-  public synchronized long createFile(TachyonURI path, TachyonURI ufsPath, long blockSizeByte,
+  public synchronized long createFile(TachyonURI path, TachyonURI ufsPath, long blockSizeBytes,
       boolean recursive) throws IOException {
     validateUri(path);
     try {
-      if (blockSizeByte > 0) {
-        return mFSMasterClient.create(path.getPath(), blockSizeByte, recursive,
-            Constants.NO_TTL);
+      if (blockSizeBytes > 0) {
+        UnderStorageType underStorageType =
+            mTachyonConf.getEnum(Constants.USER_FILE_WRITE_TYPE_DEFAULT, WriteType.class)
+                .isThrough() ? UnderStorageType.SYNC_PERSIST : UnderStorageType.NO_PERSIST;
```

Comment:
@yupeng9 I would prefer to inline this logic as it is only one statement and inlining it improves readability.